--- src/decompiled/Terraria/Item.cs
+++ src/Terraria/Terraria/Item.cs
@@ -20,6 +_,7 @@
 
 public class Item : Entity
 {
+	internal bool MystagogueBuildingModeModified = false;
 	private string _nameOverride;
 	public const int luckPotionDuration1 = 18000;
 	public const int luckPotionDuration2 = 36000;
@@ -47301,11 +_,6 @@
 			material = ItemID.Sets.IsAMaterial[type];
 
 		RebuildTooltip();
-		if (type > 0 && type < ItemID.Count && ItemID.Sets.Deprecated[type]) {
-			netID = 0;
-			type = 0;
-			stack = 0;
-		}
 	}
 
 	public void OnCreated(ItemCreationContext context)
