--- src/decompiled/Terraria/Main.cs
+++ src/Terraria/Terraria/Main.cs
@@ -68,6 +_,8 @@
 using Terraria.IO;
 using Terraria.Localization;
 using Terraria.Map;
+using Terraria.Mystagogue;
+using Terraria.Mystagogue.Commands;
 using Terraria.Net;
 using Terraria.ObjectData;
 using Terraria.Social;
@@ -80,7 +_,11 @@
 
 namespace Terraria;
 
+#if SERVER
+public class Main : Terraria.Server.Game
+#else
 public class Main : Game
+#endif
 {
 	public static class CurrentFrameFlags
 	{
@@ -5410,6 +_,9 @@
 			Map = new WorldMap(maxTilesX, maxTilesY);
 			Configuration.Load();
 			graphics = new GraphicsDeviceManager(this as Game);
+#if FNA
+			Console.WriteLine("SDL Video Driver: " + SDL3.SDL.SDL_GetCurrentVideoDriver());
+#endif
 			base.Content.RootDirectory = "Content";
 		}
 	}
@@ -15600,6 +_,11 @@
 		if (!inputTextEnter || !chatRelease)
 			return;
 
+		if (chatText.Trim().StartsWith(";") && !Anomaly.InterpretInput(chatText, true)) {
+			Anomaly.Output("You probably didn't mean to send that.");
+			return;
+		}
+
 		if (chatText != "" && !ChatManager.DebugCommands.Process((byte)myPlayer, chatText)) {
 			ChatMessage message = ChatManager.Commands.CreateOutgoingMessage(chatText);
 			if (netMode == 1)
@@ -40115,6 +_,7 @@
 	{
 		instance._mouseTextCache.isValid = false;
 		instance._mouseTextCache.noOverride = false;
+		RightClickTeleportCommand.CursorItemRecently = LocalPlayer.cursorItemIconEnabled;
 		player[myPlayer].cursorItemIconEnabled = false;
 		player[myPlayer].cursorItemIconID = 0;
 		player[myPlayer].cursorItemIconText = string.Empty;
@@ -48427,7 +_,7 @@
 					continue;
 
 				text = player[k].name;
-				if (k != myPlayer && player[myPlayer].team > 0 && player[myPlayer].team == player[k].team && netMode == 1 && player[myPlayer].HasUnityPotion() && !flag3 && !cancelWormHole) {
+				if (!flag3 && k != myPlayer && netMode == 1 && !cancelWormHole && (MapPlayerIconForceEnableCommand.PlayerMapIconFeatureFreeForceEnable || (player[myPlayer].team > 0 && player[myPlayer].team == player[k].team && player[myPlayer].HasUnityPotion()))) {
 					flag3 = true;
 					if (!unityMouseOver)
 						SoundEngine.PlaySound(12);
@@ -48465,8 +_,12 @@
 				mapFullscreen = false;
 				Vector2 vector = player[num80].Bottom - player[myPlayer].Bottom;
 				player[myPlayer].UnityTeleport(player[myPlayer].position + vector);
+				if (!MapPlayerIconForceEnableCommand.PlayerMapIconFeatureFreeForceEnable)
-				player[myPlayer].TakeUnityPotion();
+					player[myPlayer].TakeUnityPotion();
 			}
+
+			if (mapFullscreen)
+				Anomaly.AttemptMapTeleport();
 
 			cancelWormHole = false;
 			if (!flag3 && unityMouseOver)
@@ -48535,7 +_,13 @@
 
 		Player player = Main.player[myPlayer];
 		Player player2 = Main.player[playerIndex];
-		if (!player2.active || player2.ShouldNotDraw)
+		if (!player2.active)
+			return false;
+
+		if (MapPlayerIconForceEnableCommand.PlayerMapIconFeatureFreeForceEnable)
+			return true;
+
+		if (player2.ShouldNotDraw)
 			return false;
 
 		if (!player.hostile && !player2.hostile)
@@ -57150,6 +_,9 @@
 		myPlayer = 0;
 		ServerSideCharacter = false;
 		data.SetAsActive();
+
+		Mystagogue.Utils.CommandSettings.LoadAndApply(data.Player.name);
+
 		if (_pendingCharacterSelect != null) {
 			_pendingCharacterSelect();
 			_pendingCharacterSelect = null;
