--- src/decompiled/Terraria/NPC.cs
+++ src/Terraria/Terraria/NPC.cs
@@ -21,6 +_,7 @@
 using Terraria.Graphics.Effects;
 using Terraria.ID;
 using Terraria.Localization;
+using Terraria.Mystagogue.Commands;
 using Terraria.Testing;
 using Terraria.Utilities;
 using Terraria.WorldBuilding;
@@ -666,6 +_,16 @@
 			if (!spawnFriendly && RollOnlyBadLuckExtreme(50) == 0) {
 				spawnRate = (int)((float)spawnRate * 0.85f);
 				maxSpawns = (int)((float)maxSpawns * 1.15f);
+			}
+			if (player.whoAmI == Main.myPlayer && SpawnRateCommand.SpawnRate != 1) {
+				if (SpawnRateCommand.SpawnRate == 0) {
+					spawnRate = 0;
+					maxSpawns = 0;
+				}
+				else {
+					spawnRate = (int)((float)spawnRate / SpawnRateCommand.SpawnRate);
+					maxSpawns = (int)((float)maxSpawns * SpawnRateCommand.SpawnRate);
+				}
 			}
 		}
 
