--- src/decompiled/Terraria/NPC.cs
+++ src/Terraria/Terraria/NPC.cs
@@ -60927,8 +_,9 @@
 					}
 				}
 			}
-
 			bool flag15 = false;
+			NPC.spawnRate = (Mystagogue.Commands.SpawnRateCommand.SpawnRate == 0) ? 0 : NPC.spawnRate / Mystagogue.Commands.SpawnRateCommand.SpawnRate;
+			NPC.maxSpawns *= Mystagogue.Commands.SpawnRateCommand.SpawnRate;
 			if (Main.player[k].active && !Main.player[k].dead && Main.player[k].nearbyActiveNPCs < (float)maxSpawns && Main.rand.Next(spawnRate) == 0) {
 				bool flag16 = Main.player[k].ZoneTowerNebula || Main.player[k].ZoneTowerSolar || Main.player[k].ZoneTowerStardust || Main.player[k].ZoneTowerVortex;
 				spawnRangeX = (int)((double)(sWidth / 16) * 0.7);
