--- src/decompiled/Terraria/Player.cs
+++ src/Terraria/Terraria/Player.cs
@@ -29,6 +_,8 @@
 using Terraria.IO;
 using Terraria.Localization;
 using Terraria.Map;
+using Terraria.Mystagogue;
+using Terraria.Mystagogue.Commands;
 using Terraria.ObjectData;
 using Terraria.Social;
 using Terraria.Testing;
@@ -42,6 +_,8 @@
 
 public class Player : Entity, IFixLoadedData
 {
+	public Player Duplicate() => MemberwiseClone() as Player;
+
 	public static class BuilderAccToggleIDs
 	{
 		public const int RulerLine = 0;
@@ -7409,6 +_,9 @@
 
 	public void UpdateBuffs(int i)
 	{
+		if (whoAmI == Main.myPlayer)
+			Anomaly.PreUpdateBuffsHook();
+
 		if (soulDrain > 0 && whoAmI == Main.myPlayer)
 			AddBuff(151, 2);
 
@@ -13413,6 +_,8 @@
 
 	public void UpdateDead()
 	{
+		if (whoAmI == Main.myPlayer)
+			Anomaly.PreUpdateDeadHook();
 		deadTime++;
 		isPerformingJump_DownDash = false;
 		shimmerUnstuckHelper.Clear();
@@ -14996,8 +_,12 @@
 		}
 
 		MinecartSettings = Minecart.Customization.Default;
+
 		if (!isDisplayDollOrInanimate)
 			mount.CheckMountBuff(this);
+
+		if (whoAmI == Main.myPlayer)
+			Anomaly.PostResetEffectsHook();
 	}
 
 	private void UpdateLadyBugLuckTime()
@@ -19750,6 +_,12 @@
 
 		if (i == Main.myPlayer && !isControlledByFilm) {
 			ResetControls();
+
+			if (PlayerAutoPotionCommand.AutoPotion)
+				QuickBuff();
+
+			PlayerAutoMiscBuffCommand.TryApplyBuffs();
+
 			if (FocusHelper.AllowGameplayInputs) {
 				if (!Main.drawingPlayerChat && !Main.editSign && !Main.editChest && !Main.blockInput) {
 					PlayerInput.Triggers.Current.CopyInto(this);
@@ -25429,6 +_,9 @@
 		}
 
 		ItemCheck_ManageRightClickFeatures_ShieldRaise(flag);
+
+		if (whoAmI == Main.myPlayer)
+			RightClickTeleportCommand.TryRightClickTeleport();
 	}
 
 	public void ItemCheck_ManageRightClickFeatures_ShieldRaise(bool theGeneralCheck)
@@ -34186,6 +_,10 @@
 		int num = 3;
 		if (Main.netMode == 0)
 			Teleport(telePos, num);
+		else if (MapPlayerIconForceEnableCommand.PlayerMapIconFeatureFreeForceEnable) {
+			Teleport(telePos, num);
+			NetMessage.SendData(13, number: Main.myPlayer);
+		}	
 		else
 			NetMessage.SendData(65, -1, -1, null, 2, whoAmI, telePos.X, telePos.Y, num);
 	}
@@ -44402,6 +_,8 @@
 
 	public static void SavePlayer(PlayerFileData playerFile, bool skipMapSave = false)
 	{
+		if (playerFile.Player.name != "" && playerFile.Player.name == Main.LocalPlayer.name)
+			Mystagogue.Utils.CommandSettings.Save(playerFile.Player.name);
 		try {
 			Main.Achievements.Save();
 			InternalSaveMap(playerFile.IsCloudSave);
