--- src/decompiled/Terraria/UI/ItemSlot.cs
+++ src/Terraria/Terraria/UI/ItemSlot.cs
@@ -15,6 +_,8 @@
 using Terraria.ID;
 using Terraria.Initializers;
 using Terraria.Localization;
+using Terraria.Mystagogue;
 using Terraria.UI.Chat;
 using Terraria.UI.Gamepad;
 
@@ -1232,6 +_,9 @@
 		Player player = Main.player[Main.myPlayer];
 		inv[slot].newAndShiny = false;
 		if (player.itemAnimation > 0)
+			return;
+
+		if (ItemDuplicationCommand.TryItemDuplication(inv[slot]))
 			return;
 
 		if (context == 15) {
